src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForTranscript.ts

```diff
@@ -29,51 +34,82 @@ type Options = {
 };
 
 type FetchPayload = {
-  transcriptId: string;
+  genomeId: string;
   geneId: string;
+  transcriptId: string;
   options: Options;
 };
 
 export const fetchForTranscript = async (payload: FetchPayload) => {
   const {
+    genomeId,
     geneId,
     transcriptId,
     options: { transcript: transcriptOptions, gene: geneOptions }
   } = payload;
-  const urls = buildUrlsForTranscript(transcriptId, transcriptOptions);
+  const productGeneratingContext = await fetchSequenceChecksums({
+    genomeId,
+    transcriptId
+  });
+  const urls = buildUrlsForTranscript(
+    { geneId, productGeneratingContext },
+    transcriptOptions
+  );
+
   if (geneOptions.genomicSequence) {
-    urls.push(buildFetchUrl(geneId, 'genomicSequence'));
+    urls.push(buildFetchUrl({ geneId }, 'genomicSequence'));
   }
+
   const sequencePromises = urls.map((url) =>
     fetch(url).then((response) => response.text())
   );
   const sequences = await Promise.all(sequencePromises);
-  const combinedFasta = sequences.join('\n');
+  const combinedFasta = sequences.join('\n\n');
 
   downloadAsFile(combinedFasta, `${transcriptId}.fasta`, {
     type: 'text/x-fasta'
   });
 };
 
 const buildUrlsForTranscript = (
-  id: string,
+  data: {
+    geneId: string;
+    productGeneratingContext: ProductGeneratingContextFragment;
+  },
   options: Partial<TranscriptOptions>
 ) => {
   return options
     ? transcriptOptionsOrder
         .filter((option) => options[option])
-        .map((option) => buildFetchUrl(id, option))
+        .map((option) => buildFetchUrl(data, option))
     : [];
 };
 
-const buildFetchUrl = (id: string, sequenceType: TranscriptOption) => {
-  const sequenceTypeToTypeParam: Record<TranscriptOption, string> = {
+const buildFetchUrl = (
+  data: {
+    geneId: string;
+    productGeneratingContext?: ProductGeneratingContextFragment;
+  },
+  sequenceType: TranscriptOption
```

Comment:
- If we are transforming the first parameter into an object; I think we should include the second parameter in this object as well, rather than have it as an independent string
- The name `productGeneratingContext` is awkward. Let's change the name to `checksums`, and the object representing it to:

```
{
  cdna: string;
  cds: string;
  product: string;
}
```

notice that for non-coding transcripts we won't be able to get the cDNA at the moment, because Thoas doesn't currently return product-generating contexts for them:

![image](https://user-images.githubusercontent.com/6834224/106130632-7f42cd00-6159-11eb-9c05-b99297ff47dd.png)

Something to be discussed w/Steve and Andrea.

Reply:
P.S.: I am also not sure it's a good idea to call a protein a product.

At the thinking about CDM currently stands, Thoas will return either proteins or mature RNAs as products. The user in the UI is requesting a protein; so I suppose we should be explicit in our code that such-and-such code path is for protein.

Reply:
Will this do for now: 9a42f55?